mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -0,0 +1,122 @@
+    for (Element child : classElement.getEnclosedElements()) {
+      if (child.getKind() == ElementKind.METHOD) {
+        ExecutableElement method = (ExecutableElement) child;
+        String methodName = method.getSimpleName().toString();
+        if (methodName.startsWith("get") && method.getParameters().isEmpty()) {
+          TypeName propertyType = TypeName.get(method.getReturnType());
+          if (TypeName.VOID.equals(propertyType)) {
+            continue;
+          }
+          String propertyName = Introspector.decapitalize(methodName.substring(3));
+          DefaultPropertyDefinition.Builder builder = propertyBuilders.get(propertyName);
+          if (builder == null) {
+            builder = new DefaultPropertyDefinition.Builder(propertyName, propertyType);
+            propertyBuilders.put(propertyName, builder);
+          } else if (!builder.getType().equals(propertyType)) {
+            context
+                .getMessager()
+                .warn(
+                    method,
+                    "Ignoring method %s %s() because there is a setter "
+                        + "with the same name but a different type: %s(%s)",
+                    propertyType,
+                    methodName,
+                    builder.getSetterName(),
+                    builder.getType());
+            continue;
+          }
+          builder.withGetterName(methodName);
+        } else if (methodName.startsWith("set") && method.getParameters().size() == 1) {
+          String propertyName = Introspector.decapitalize(methodName.substring(3));
+          VariableElement parameter = method.getParameters().get(0);
+          TypeName propertyType = TypeName.get(parameter.asType());
+          DefaultPropertyDefinition.Builder builder = propertyBuilders.get(propertyName);
+          if (builder == null) {
+            builder = new DefaultPropertyDefinition.Builder(propertyName, propertyType);
+            propertyBuilders.put(propertyName, builder);
+          } else if (!builder.getType().equals(propertyType)) {
+            context
+                .getMessager()
+                .warn(
+                    method,
+                    "Ignoring method %s(%s) because there is a getter "
+                        + "with the same name but a different type: %s %s()",
+                    methodName,
+                    propertyType,
+                    builder.getType(),
+                    builder.getGetterName());
+            continue;
+          }
+          builder.withSetterName(methodName);
+        }
+      }
+    }
+
+    ImmutableList.Builder<PropertyDefinition> definitions = ImmutableList.builder();
+    for (DefaultPropertyDefinition.Builder builder : propertyBuilders.values()) {
+      if (builder.getGetterName() != null && builder.getSetterName() != null) {
+        definitions.add(builder.build());
+      }
+    }
+
+    String entityName = Introspector.decapitalize(classElement.getSimpleName().toString());
```

Comment:
Yes, the table name is directly derived from the entity for now. [JAVA-2110](https://datastax-oss.atlassian.net/browse/JAVA-2110) will introduce naming strategies. We can also add a way to override it through annotations (either as an attribute of `@Entity`, or as another annotation).

If the table name is injected from the DAO factory method, it will override anything that is defined at the DAO level. The generated code will determine that in `initAsync`, and prepare the statements to target the appropriate keyspaces: for statements generated with the query builder, it's trivial to substitute; for user-provided query strings, we'll adopt the convention that the query string must contain a placeholder (e.g. `${table}`), and the generated code will do a string substitution right before preparing.